opni-insights-service/app/main.py

```diff
@@ -399,3 +400,51 @@ async def index_logs(start_ts: int, end_ts: int):
     except Exception as e:
         # Bad Request
         logging.error(e)
+
+
+@app.get("/areas_of_interest")
+async def get_areas_of_interest(start_ts: int, end_ts: int):
+    logging.info(
+        f"Received request to obtain all areas of interest between {start_ts} and {end_ts}"
+    )
+
+    query = {
+        "query": {
+            "bool": {
+                "must": [{"match": {"anomaly_level.keyword": "Suspicious"}}],
+                "filter": [{"range": {"timestamp": {"gte": start_ts, "lte": end_ts}}}],
+            }
+        },
+        "aggs": {
+            "logs_over_time": {
+                "date_histogram": {"field": "timestamp", "calendar_interval": "minute"},
+                "aggs": {"total_normal": {"sum": {"field": "anomaly_predicted_count"}}},
+            }
+        },
+    }
+    res = await es_instance.search(index="logs", body=query, size=0)
+    df = pd.DataFrame(res["aggregations"]["logs_over_time"]["buckets"])
+    df["moving_avg"] = df["doc_count"].rolling(60).mean() * 1.3
+    df["AOI"] = df.doc_count.gt(df.moving_avg.shift())
+    areas_of_interest_df = df[df.AOI == True]
+    areas_of_interest_df["time_diff"] = areas_of_interest_df["key"].diff()
+
+    areas_of_interest = []
+
+    current_aoi_start = -1
+    prev_segment_start = -1
+    for index, row in areas_of_interest_df.iterrows():
+        if current_aoi_start == -1:
+            current_aoi_start = row["key"]
+        if row["time_diff"] != 60000.0:
+            areas_of_interest.append(
+                {"start_ts": current_aoi_start, "end_ts": prev_segment_start + 60000}
```

Comment:
There needs to be logic added to handle the case where prev_segment_start is also equal to -1. Otherwise, if it is -1 when this if block is hit, the end_ts will be set to 59999. I came across this when the start_ts was set to 1632788580000 and the end_ts was set to 1633111662322

Reply:
Addressed